lib/bamboo/adapters/mailgun_adapter.ex

```diff
@@ -0,0 +1,125 @@
+defmodule Bamboo.MailgunAdapter do
+  @moduledoc """
+  Sends email using Mailgun's API.
+
+  Use this adapter to send emails through Mailgun's API. Requires that an API
+  key and a domain are set in the config.
+
+  ## Example config
+
+      # In config/config.exs, or config.prod.exs, etc.
+      config :my_app, MyApp.Mailer,
+        adapter: Bamboo.MailgunAdapter,
+        api_key: "my_api_key",
+        domain: "your.domain"
+
+      # Define a Mailer. Maybe in lib/my_app/mailer.ex
+      defmodule MyApp.Mailer do
+        use Bamboo.Mailer, otp_app: :my_app
+      end
+  """
+
+  @base_uri "https://api.mailgun.net/v3/"
+  @behaviour Bamboo.Adapter
+
+  alias Bamboo.Email
+
+  defmodule ApiError do
+    defexception [:message]
+
+    def exception(%{params: params, response: response}) do
+      message = """
```

Comment:
Can you take a look at https://github.com/thoughtbot/bamboo/blob/master/lib/bamboo/adapters/mandrill_adapter.ex#L30 and the corresponding test https://github.com/thoughtbot/bamboo/blob/master/test/lib/bamboo/adapters/mandrill_adapter_test.exs#L140

I think we should do the same for this adapter so the API key isn't in the logs